src/main/java/com/minguard/controller/IncidentController.java

```diff
@@ -0,0 +1,100 @@
+package com.minguard.controller;
+
+import com.minguard.dto.incident.IncidentResponse;
+import com.minguard.dto.incident.RegisterIncidentResponse;
+import com.minguard.dto.incident.UpdateIncidentRequest;
+import com.minguard.dto.incident.RegisterIncidentRequest;
+import com.minguard.entity.Incident;
+import com.minguard.entity.User;
+import com.minguard.mapper.IncidentMapper;
+import com.minguard.service.spec.IncidentService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.NotImplementedException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+@RequiredArgsConstructor
+@Tag(name = "Incident Controller")
+@RequestMapping("/incident")
+@RestController
+public class IncidentController {
+
+    private final IncidentService incidentService;
+
+    
+    @PreAuthorize("hasAuthority('ADMIN')")
+    @GetMapping("/{incidentId}")
+    @Operation(summary = "Get incident by id", description = "Get details of the incident by id.")
+    public ResponseEntity<IncidentResponse> incidentById(@PathVariable Long incidentId) {
+        Incident incident = incidentService.getIncidentById(incidentId);
+
+        return ResponseEntity.status(HttpStatus.OK).body(IncidentMapper.INSTANCE.toResponse(incident));
+    }
+
+    @PreAuthorize("hasAuthority('ADMIN')")
+    @GetMapping
+    @Operation(summary = "Get all incidents", description = "Get details of all incidents. Must have admin role.")
+    public ResponseEntity<List<IncidentResponse>> allIncidents() {
+        List<IncidentResponse> incidents = incidentService.findAll();
+
+        return ResponseEntity.status(HttpStatus.OK).body(incidents);
+    }
+
+    @PreAuthorize("hasAuthority('ADMIN')")
```

Comment:
Acho que esse vai ser usado pelo app também
Se for pra listagem, podemos pensar se compensa retornar todos os dados de todos os incidentes ou alguns dados principais e, ao clicar em cima, buscar todos os dados por id. Se forem poucos incidentes talvez a diferença não seja tão significativa